new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleNginxValidation.java

```diff
@@ -290,76 +303,115 @@ public void testCreateMiiDomain() {
+
+      logger.info("ingress {0} for cluster {1} of domain {2} is created in namespace {3}",
+          ingressName, clusterName, domainUid, domainNamespace);
+    }
   }
 
   @Test
   @Order(3)
-  @DisplayName("Verify the application can be accessed through the ingress controller")
+  @DisplayName("Verify the application can be accessed through the ingress controller for each cluster in the domain")
   public void testSampleAppThroughIngressController() {
 
-    List<String> managedServerNames = new ArrayList<>();
-    for (int i = 1; i <= replicaCount; i++) {
-      managedServerNames.add(managedServerNameBase + i);
+    for (int i = 1; i <= NUMBER_OF_CLUSTERS; i++) {
+      String clusterName = CLUSTER_NAME_PREFIX + i;
+
+      expectedServerNamesInAppResponse.clear();
+      for (int j = 1; j <= replicaCount; j++) {
+        expectedServerNamesInAppResponse.add(clusterName + managedServerNameBase + j);
+      }
+
+      // check that NGINX can access the sample apps from all managed servers in the cluster of the domain
+      curlCmd = getCurlCmd(clusterName);
+      assertThat(callWebAppAndCheckForServerNameInResponse(curlCmd, expectedServerNamesInAppResponse, 50))
+          .as("verify NGINX can access the sample app from all managed servers in the domain")
+          .withFailMessage("NGINX can not access the sample app from one or more of the managed servers")
+          .isTrue();
     }
+  }
 
-    // check that NGINX can access the sample apps from all managed servers in the domain
-    String curlCmd = String.format("curl --silent --noproxy '*' -H 'host: %s' http://%s:%s/sample-war/index.jsp",
-        domainUid + ".test", K8S_NODEPORT_HOST, nodeportshttp);
-    assertThat(callWebAppAndCheckForServerNameInResponse(curlCmd, managedServerNames, 50))
-        .as("NGINX can access the sample app from all managed servers in the domain")
-        .withFailMessage("NGINX can not access the sample app from one or more of the managed servers")
-        .isTrue();
+  @Test
+  @Order(4)
+  @DisplayName("Verify scale domain with domainUid with multiple clusters in domainNamespace")
+  public void testScaleDomainWithMultiClusters() {
+
+    for (int i = 1; i <= NUMBER_OF_CLUSTERS; i++) {
+      String clusterName = CLUSTER_NAME_PREFIX + i;
+
+      // set the expected server name list which should return by the app response before scale
+      expectedServerNamesInAppResponse.clear();
+      for (int j = 1; j <= replicaCount; j++) {
+        expectedServerNamesInAppResponse.add(clusterName + managedServerNameBase + j);
+      }
+      logger.info("expected server name list which should be in the sample app response: {0} before scale",
+          expectedServerNamesInAppResponse);
+
+      // get a random integer between [0 - 5]
+      int numberOfServers = new Random().nextInt(6);
+      logger.info("scaling the cluster {0} of domain {1} in namespace {2} to {3} servers",
+          clusterName, domainUid, domainNamespace, numberOfServers);
```

Comment:
IMO this kind of randomness belongs more in a stress test than a most QA tests....  Is this randomness deliberate? It means repeatability between test runs is gone - requiring a change to the test source code to make it repeatable when debugging.

Reply:
Also, 5 servers seems like a lot on the surface for this kind of test - it potentially slows the test down forever more, particular on systems that might not have enough CPU/memory to handle that many servers quickly.

Reply:
You are right. Changed the logic a little bit.

Reply:
LGTM